### PR TITLE
[AssetMapper] Split test dirs in tests

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php [hunk 1]
```diff
@@ -41,19 +41,24 @@ public function testGetStorageDir()
...
         $this->expectException(\RuntimeException::class);
-        $this->expectExceptionMessage('file_put_contents('.$vendorDir.'/module_specifier/module_specifier.index.js): Failed to open stream: No such file or directory');
-        $storage->save($entry, 'any content');
+        $this->expectExceptionMessage('file_put_contents('.$vendorDir.'/module_specifier/module_specifier.index.js): Failed to open stream: Permission denied');
 
-        $this->filesystem->remove($vendorDir);
+        try {
+            $storage->save($entry, 'any content');
+        } finally {
+            $this->filesystem->chmod($vendorDir.'/module_specifier/module_specifier.index.js', 0777);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php:60]: this is the fix: revert the chmod before remove()
/cc @xabbuh @smnandre FYI

**Review by GromNaN — APPROVED** [review]
Why not. Have you noticed a problem when using the same dir?

**Comment by nicolas-grekas** [comment]
Yes: this creates coupling between test cases. If one test fails, many others fail in cascade. This cost us several hours of debugging.

File: src/Symfony/Component/AssetMapper/Tests/Path/LocalPublicAssetsFilesystemTest.php [hunk 2]
```diff
@@ -18,12 +18,12 @@
...
 {
     private Filesystem $filesystem;
-    private static string $writableRoot = __DIR__.'/../Fixtures/importmaps_for_writing';
+    private static string $writableRoot = __DIR__.'/../Fixtures/local_public_assets_filesystem';
 
     protected function setUp(): void
     {
         $this->filesystem = new Filesystem();
-        if (!file_exists(__DIR__.'/../Fixtures/importmaps_for_writing')) {
+        if (!file_exists(__DIR__.'/../Fixtures/local_public_assets_filesystem')) {
```
Comment (xabbuh) [inline src/Symfony/Component/AssetMapper/Tests/Path/LocalPublicAssetsFilesystemTest.php:26]:
```suggestion
        if (!file_exists(self::writableRoot)) {
```

Comment (nicolas-grekas) [reply src/Symfony/Component/AssetMapper/Tests/Path/LocalPublicAssetsFilesystemTest.php:26]: oops sorry I merged before seeing your comment

File: src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php (hunk 1 above)
Comment (smnandre) [reply src/Symfony/Component/AssetMapper/Tests/ImportMap/RemotePackageStorageTest.php:60]: Well...  💡   that makes sense.

**Pull request merged**